app/classes/suggestion.rb

```diff
@@ -77,15 +77,33 @@ def name_with_most_observations(names)
     best_name
   end
 
+  # Nimmo note: rails t -n /ObserverControllerTest#test_suggestions/
   def example_image_obs
-    id = Observation.connection.select_value(%(
-      SELECT o.id FROM observations o
-      JOIN images i ON i.id = o.thumb_image_id
-      WHERE o.name_id IN (#{@name.synonym_ids.join(",")})
-      AND o.vote_cache >= 2
-      ORDER BY o.vote_cache + i.vote_cache DESC
-      LIMIT 1
-    ))
-    Observation.safe_find(id)
+    # id = Observation.connection.select_value(%(
+    #   SELECT o.id FROM observations o
+    #   JOIN images i ON i.id = o.thumb_image_id
+    #   WHERE o.name_id IN (#{@name.synonym_ids.join(",")})
+    #   AND o.vote_cache >= 2
+    #   ORDER BY o.vote_cache + i.vote_cache DESC
+    #   LIMIT 1
+    # ))
+
+    # Joining the Arel tables explicitly on(x) generates a clean INNER JOIN:
+    #
+    # SELECT `observations`.`id` FROM `observations`
+    # INNER JOIN `images` ON `images`.`id` = `observations`.`thumb_image_id`
+    # WHERE (`observations`.`name_id` = 981143639)
+    # AND (`observations`.`vote_cache` >= 2.0)
+    # ORDER BY (`observations`.`vote_cache` + `images`.`vote_cache`) DESC
+    # LIMIT 1
+
+    o_i_inner_join = Observation.arel_table.join(Image.arel_table).
+                     on(Image[:id].eq(Observation[:thumb_image_id]))
+
+    # Below we just return the observation itself, not the id then safe_find(id)
+    # Note: Ruby 2.7 will allow endless ranges like (vote_cache: 2..)
+    Observation.joins(o_i_inner_join.join_sources).
+      where(name_id: @name.synonym_ids, vote_cache: 2..Float::INFINITY).
+      order((Observation[:vote_cache] + Image[:vote_cache]).desc).take
   end
```

Comment:
Here's what I found:
> The take method returns nil if no record is found and no exception will be raised.
> The [take!](https://api.rubyonrails.org/v7.0.3/classes/ActiveRecord/FinderMethods.html#method-i-take-21) method behaves exactly like take, except that it will raise ActiveRecord::RecordNotFound if no matching record is found.

Reply:
By the way - The actual SQL generated by this statement is in pasted into the code comments currently.
It's essentially the same, the `take` generates a `LIMIT 1` at the end.

Reply:
Also... I can't find `example_image_obs` used anywhere in MO. Maybe it's retired!

Reply:
Oh sorry, it's used in `image_obs`, which is used in `observer/suggestions.html.erb`.

Reply:
Ok. The old method used `safe_find(id)` and has a `rescue` clause.
I think that means we want to use `take!` instead:
```
        def safe_find(id)
          find(id)
        rescue ::ActiveRecord::RecordNotFound
          nil
        end
```

Reply:
Sorry it took me a bit to understand your question!  I'm committing the change from `take` to `take!` and removing extra comments that had the SQL for comparison.  

I believe the method now preserves original behavior to the letter.

Reply:
What??  Safe_find *protects* us from the exception raised by straight-up find.  That means we do *not* want this code crashing if nothing matches.  That means using take -- just like you had it -- is what we want.  Your latest version using take! will crash if nothing matches.

(Being a die-hard perl programmer, I detest code that crashes when it fails.  Failed code should return nil, not crash.  Crashing is *never* the behavior I want.  Never, ever, ever.  Returning nil might not be right in some cases, either, but 90% of the time it is just fine.  Crashing is right 0% of the time.  I'd much rather the webpage display an empty result than crash and thereby hide the entire page and give the user nowhere to go or even a clue as to what might have gone wrong.  I get the principle that good code should take into account the possibility of failure and handle it appropriately and all.  But throwing an exception in the case of failure is a bit like rapping a student's knuckles when they get an answer wrong: just sort of sadistic and counter-productive!  Ahem.  Sorry for the rant.  Just sort of frustrating to me.  All the more so because people who defend the crashing behavior -- which seems to be most programmers these days -- pretend that they have some sort of moral high ground, but I respectfully disagree.)

By the way, this code is only used by the image recognition alpha-testing code which is only seen by admins.  And since I have failed to figure out how to install Alan Celestino's model on the new image server, it's been broken for several months.  That means it's not actually seen by *anyone*.  Just sayin'.  Consider this low priority! :)

Reply:
Ahhhh.... OK! Will restore `take`